### PR TITLE
Extrapolate Jaeger transaction count from reported sampling rate

### LINKED ISSUE #3011: [Jaeger] How to support sampling
When using probabilistic sampling Jaeger only sends sampled spans. Elastic APM UI currently requires a 100% of the transaction events. Elastic APM agent only sample spans.
Investigate how to solve sampling with Jaeger integration.

Part of elastic/apm-server#2886

UPDATE:
As probabilistic sampling support was added with elastic/apm-server#3487 this Issue now solely focuses on changes necessary to integrate Jaeger sampled traces into the APM app.

### PR DESCRIPTION
**Pull request by axw**
## Motivation/summary

When Jaeger is configured to sample a percentage of traces, then the statistics reported in APM UI will be proportional to the sampling rate, and not the actual number of operations. Jaeger reports the sampling rate as a pair of tags (sampler.type and sampler.param) with each span. We can use these to extrapolate the number of transactions when performing aggregations.

## Checklist

- [x] I have signed the [Contributor License Agreement](https://www.elastic.co/contributor-agreement/).
- [x] My code follows the style guidelines of this project (run `make check-full` for static code checks and linting)
- [x] I have rebased my changes on top of the latest master branch
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing [**unit** tests](https://github.com/elastic/apm-server/blob/master/TESTING.md) pass locally with my changes
- [x] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)

## How to test these changes

1. Enable transaction aggregation (`apm-server.aggregation.enabled=true`)
2. Run an app instrumented with Jaeger, configured with remote sampling
3. Set a sampling rate of 0.1 for the service in APM UI
4. Run 1000 operations
5. Check there are 100 (i.e. 1000 * 0.1) transactions
6. Check the sum of counts in `transaction.duration.histogram` fields is 1000 (e.g. using https://github.com/elastic/elasticsearch/pull/55933 when it lands)

## Related issues

Closes #3011

### CODE REVIEW

**Comment by axw** [comment]
Rounding error on this is going to be pretty bad for some sampling rates (e.g 0.3, 0.4), so it might be worth scaling all counts recorded in the in-memory histograms up when recording (e.g. multiply by 100), and then back down when creating metricset docs.

File: x-pack/apm-server/aggregation/txmetrics/aggregator_test.go [hunk 1]
```diff
@@ -219,6 +219,71 @@ func TestAggregatorRunPublishErrors(t *testing.T) {
...
+	}
+
+	stopAggregator := runAggregator(agg)
+	defer stopAggregator()
+
+	req := expectPublish(t, reqs)
+	require.Len(t, req.Transformables, 1)
+	metricset := req.Transformables[0].(*model.Metricset)
+	require.Len(t, metricset.Samples, 1)
+	assert.Equal(t, []int64{3 /*round(1+1.5)*/}, metricset.Samples[0].Counts)
```
Comment (simitt) [inline x-pack/apm-server/aggregation/txmetrics/aggregator_test.go:284]: Do you mind adding a comment explaining this check for `metricset`? I find it hard to follow at the end of this test.

Comment (axw) [reply x-pack/apm-server/aggregation/txmetrics/aggregator_test.go:288]: I've added a comment. If it's still unclear let me know, I'm happy to do a follow up.
